modules/nomad-cluster/main.tf

```
@@ -136,6 +136,7 @@ resource "aws_security_group" "lc_security_group" {
 }
 
 resource "aws_security_group_rule" "allow_ssh_inbound" {
+  count       = length(var.allowed_ssh_cidr_blocks) >= 1 ? 1 : 0
```

NIT: this is a bit more idiomatic:

```suggestion
  count       = length(var.allowed_ssh_cidr_blocks) > 0 ? 1 : 0
```